internal/controller/bucket/delete.go

```diff
@@ -68,22 +68,51 @@ func (c *external) Delete(ctx context.Context, mg resource.Managed) error {
 		})
 	}
 
-	if err := g.Wait(); err != nil {
-		return errors.Wrap(err, errDeleteBucket)
+	// Wait for all go routines to finish
+	deleteErr := g.Wait()
+
+	// Regardless of whether an error occurred, we need to update the Bucket CR Status
+	// to cover the following scenarios:
+	// 1. The caller attempts to delete the CR and an error occurs during the call to
```

Comment:
Nits: these comments are duplicated with https://github.com/linode/provider-ceph/blob/765853504a7a94d5ac1f9a5a00a564d3b5245a59/internal/controller/bucket/delete.go#L27-L37.
I think you can merge comments in those L27-L37 to this block

Reply:
Thanks. I must've messed this up during a rebase :+1: